src/Compilers/CSharp/Test/Syntax/Parsing/CollectionExpressionParsingTests.cs

```diff
@@ -18758,4 +18758,1991 @@ public void CollectionExpression_KeywordKey_Illegal5()
+            }
+            N(SyntaxKind.CloseBracketToken);
+        }
+        EOF();
+    }
+
+    [Fact]
+    public void WithElement37()
+    {
+        UsingExpression("[with()++]",
+            // (1,8): error CS1003: Syntax error, ',' expected
+            // [with()++]
+            Diagnostic(ErrorCode.ERR_SyntaxError, "++").WithArguments(",").WithLocation(1, 8),
+            // (1,10): error CS1525: Invalid expression term ']'
+            // [with()++]
+            Diagnostic(ErrorCode.ERR_InvalidExprTerm, "]").WithArguments("]").WithLocation(1, 10));
+
+        N(SyntaxKind.CollectionExpression);
+        {
+            N(SyntaxKind.OpenBracketToken);
+            N(SyntaxKind.WithElement);
+            {
+                N(SyntaxKind.WithKeyword);
+                N(SyntaxKind.ArgumentList);
+                {
+                    N(SyntaxKind.OpenParenToken);
+                    N(SyntaxKind.CloseParenToken);
+                }
+            }
+            M(SyntaxKind.CommaToken);
+            N(SyntaxKind.ExpressionElement);
+            {
+                N(SyntaxKind.PreIncrementExpression);
+                {
+                    N(SyntaxKind.PlusPlusToken);
+                    M(SyntaxKind.IdentifierName);
+                    {
+                        M(SyntaxKind.IdentifierToken);
+                    }
+                }
+            }
+            N(SyntaxKind.CloseBracketToken);
+        }
+        EOF();
+    }
+
+    [Fact]
+    public void WithElement38()
+    {
+        UsingExpression("[with()[0]]",
+            // (1,8): error CS1003: Syntax error, ',' expected
+            // [with()[0]]
+            Diagnostic(ErrorCode.ERR_SyntaxError, "[").WithArguments(",").WithLocation(1, 8));
+
+        N(SyntaxKind.CollectionExpression);
+        {
+            N(SyntaxKind.OpenBracketToken);
+            N(SyntaxKind.WithElement);
+            {
+                N(SyntaxKind.WithKeyword);
+                N(SyntaxKind.ArgumentList);
+                {
+                    N(SyntaxKind.OpenParenToken);
+                    N(SyntaxKind.CloseParenToken);
+                }
+            }
+            M(SyntaxKind.CommaToken);
+            N(SyntaxKind.ExpressionElement);
+            {
+                N(SyntaxKind.CollectionExpression);
+                {
+                    N(SyntaxKind.OpenBracketToken);
+                    N(SyntaxKind.ExpressionElement);
+                    {
+                        N(SyntaxKind.NumericLiteralExpression);
+                        {
+                            N(SyntaxKind.NumericLiteralToken, "0");
+                        }
+                    }
+                    N(SyntaxKind.CloseBracketToken);
+                }
+            }
+            N(SyntaxKind.CloseBracketToken);
+        }
+        EOF();
+    }
+
+    [Fact]
+    public void WithElement39()
+    {
+        UsingTree("""
+            void M()
+            {
+                var v = [await with()];
```

Comment:
>with() [](http://example.com/codeflow?start=31&length=6)

Should this be parsed as an invocation expression, rather than as a syntax error?